shared/qcommon/q_platform.h

```diff
@@ -227,11 +227,76 @@ along with this program; if not, see <http://www.gnu.org/licenses/>.
 #endif
 
 // endianness
-void CopyShortSwap( void *dest, void *src );
-void CopyLongSwap( void *dest, void *src );
-short ShortSwap( short l );
-int LongSwap( int l );
-float FloatSwap( const float *f );
+// Use compiler builtins where possible for maximum performance
+#include <stdint.h>
+#if defined(__GNUC__) || defined(__GNUG__)
```

Comment:
I think this needs to be `#if !defined(__clang__) && (defined(__GNUC__) || defined(__GNUG__))` otherwise for clang it will enter this block, and then not define the functions. Do you agree?